sphinx/texinputs/sphinx.sty

```diff
@@ -282,6 +255,72 @@
   % LaTeX environments always revert local changes on exit, here e.g. \parskip
 }
 
+% For shadowing boxes, with allowed page breaks
+\def\Sphinx@shadowfboxsep {5\p@}% 5pt used to be hard-coded in latex.py
+\def\Sphinx@shadowsize    {4\p@}% 4pt was the default for \shadowbox from fancybox.sty
+
+\long\def\Sphinx@ShadowFBox#1{%
+  \leavevmode\begingroup
+  \setbox\@tempboxa
+   \hbox{\vrule\@width\fboxrule
+         \vbox{\hrule\@height\fboxrule
+               \kern\Sphinx@shadowfboxsep
+               \hbox{\kern\Sphinx@shadowfboxsep #1\kern\Sphinx@shadowfboxsep}%
+               \kern\Sphinx@shadowfboxsep
+               \hrule\@height\fboxrule}%
+         \vrule\@width\fboxrule }%
+  % we imitate what \shadowbox from fancybox.sty would do
+  \dimen@\dimexpr.5\fboxrule+\Sphinx@shadowsize\relax
+  \hbox{\vbox{\offinterlineskip
+        \hbox{\copy\@tempboxa
+              \kern-.5\fboxrule
+              \lower\Sphinx@shadowsize
+              \hbox{\vrule\@height\ht\@tempboxa \@width\dimen@}}%
+        \kern-\dimen@
+        \moveright\Sphinx@shadowsize\vbox{\hrule\@width\wd\@tempboxa \@height\dimen@}%
+        }\kern-\Sphinx@shadowsize % shadow on right does not add to width
+       }%
+  \endgroup
+}
+
+\newenvironment{SphinxShadowBox}
+  {\def\FrameCommand {\Sphinx@ShadowFBox }%
+   % earlier use from latex.py  of "center" and a \vbox was buggy and caused
+   % vertical space which we try to imitate as much as possible via "trivlist"
+   \OuterFrameSep \z@skip
+   \trivlist\item\relax\null % makes blank line on top to imitate <= 1.4.1
+   \MakeFramed {\FrameRestore
+    % trick LaTeX into thinking we are in a minipage, becauser earlier Sphinx
+    % used a minipage (from latex.py) and this affects spacing in list.
+    % This is a copy-paste from LaTeX source code.
+    \@pboxswfalse
+    % the "\width" here means width from the frame added by \Sphinx@ShadowFBox
+    \hsize\dimexpr\linewidth-\width\relax \textwidth\hsize \columnwidth\hsize
+    \@parboxrestore
+       % for footnotes, but Sphinx inactivates footnotes in topics
+       \def\@mpfn{mpfootnote}\def\thempfn{\thempfootnote}\c@mpfootnote\z@
+       \let\@footnotetext\@mpfootnotetext
+    \let\@listdepth\@mplistdepth \@mplistdepth\z@
+    \@minipagerestore
+    \@setminipage
+    }%
+  }%
+  {\par\unskip
+   % handles footnotes
+   \ifvoid\@mpfootins\else
+      \vskip\skip\@mpfootins\normalcolor\footnoterule\unvbox\@mpfootins
+   \fi
+   \@minipagefalse
+   \endMakeFramed
+   % a subtlety here. It appears that if @inlabel is true, there will be extra
+   % blank line inserted, which we don't want. Happens when topic contents is
+   % not a list. But on the other hand for the tables of contents boxes, we do
+   % want extra blank line to get same result as with Sphinx <= 1.4.1. Turns
+   % out the \null does it.
+   \if@inlabel\global\@inlabelfalse\global\@newlistfalse\else\null\fi
```

Comment:
This is the check mentioned above. Except if the material handled in between toggled \@inlabel switch, LaTeX will think nothing was found after \item, because no \par was done (after \endMakeFramed).